include/dxc/HLSL/DxilSigPoint.inl

```diff
@@ -50,46 +50,44 @@ const SigPoint SigPoint::ms_SigPoints[kNumSigPointRecords] = {
+  {/*DepthLessEqual*/         D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(NA),         D(NA),           D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NotPacked _50), D(NA)}, \
+  {/*DepthGreaterEqual*/      D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(NA),         D(NA),           D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NotPacked _50), D(NA)}, \
+  {/*StencilRef*/             D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(NA),         D(NA),           D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NotPacked _50), D(NA)}, \
+  {/*DispatchThreadID*/       D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(NA),         D(NA),           D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NA),            D(NotInSig)}, \
+  {/*GroupID*/                D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(NA),         D(NA),           D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NA),            D(NotInSig)}, \
+  {/*GroupIndex*/             D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(NA),         D(NA),           D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NA),            D(NotInSig)}, \
+  {/*GroupThreadID*/          D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(NA),         D(NA),           D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NA),            D(NotInSig)}, \
+  {/*TessFactor*/             D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(TessFactor), D(TessFactor),   D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NA),            D(NA)}, \
+  {/*InsideTessFactor*/       D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(TessFactor), D(TessFactor),   D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NA),            D(NA),            D(NA)}, \
+  {/*ViewID*/                 D(NotInSig _61), D(NA),    D(NotInSig _61), D(NotInSig _61), D(NA),     D(NA),      D(NA),         D(NotInSig _61), D(NA),     D(NA),    D(NA),    D(NotInSig _61), D(NA),    D(NotInSig _61),  D(NA),            D(NA)}, \
+  {/*Barycentrics*/           D(NA),           D(NA),    D(NA),           D(NA),           D(NA),     D(NA),      D(NA),         D(NA),           D(NA),     D(NA),    D(NA),    D(NA),           D(NA),    D(NotPacked _61), D(NA),            D(NA)},
 // INTERPRETATION-TABLE:END
```

Comment:
Modifying this table directly is incorrect.  It is generated by running hctdb_instrhelp.py -update-files.
I thought the nested macro problem was due to name collision between a macro name (DO) and macro parameter name (was DO, changed to ROW) before.  This version should have fixed that issue.  Does it still fail?  What is the problem that remained with the expansion in this version exactly?

Reply:
What do you think @ehsannas?

Reply:
I haven't change the python script that generates this code (I wasn't very familiar with it).
The issue is not about name collision. The issue stems from early expansion vs. late expansion of macros, depending on the compiler.

For example:
```
 define _AB 1,2
 define SEC(args) (args)
 define FIRST(args) SEC(args)
  FIRST(_AB);

A preprocessor that delays expansion of _AB will expand thus:

FIRST(_AB) => SEC(_AB) => (_AB) => (1,2)

whereas an early expanding preprocessor will expand thus:

FIRST(_AB) => SEC(_AB) => SEC(1,2) => ERROR! SEC only takes 1 arg!
```

We had to make a similar change (https://github.com/Microsoft/DirectXShaderCompiler/pull/1270) before. There is a much more complete explanation there.

Reply:
Here's an example of errors that I got:

```
In file included from /usr/local/google/home/ehsann/DirectXShaderCompiler/lib/HLSL/DxilSignature.cpp:241:
../include/dxc/HLSL/DxilSigPoint.inl:92:3: error: too many arguments provided to function-like macro invocation
  DO_INTERPRETATION_TABLE(DO_ROW)
  ^
../include/dxc/HLSL/DxilSigPoint.inl:66:173: note: expanded from macro 'DO_INTERPRETATION_TABLE'
  ROW(SampleIndex,            NA,           NA,    NA,           NA,           NA,     NA,      NA,         NA,           NA,     NA,    NA,    NA,           NA,    Shadow _41,    NA,            NA) \
                                                                                                                                                                            ^
../include/dxc/HLSL/DxilSigPoint.inl:86:14: note: expanded from macro '_41'
#define _41 ,4,1
             ^
../include/dxc/HLSL/DxilSigPoint.inl:89:9: note: macro 'DO' defined here
#define DO(k) VersionedSemanticInterpretation(DXIL::SemanticInterpretationKind::k)
        ^
```

I see several such errors (1 per row I suppose).

Reply:
Thanks for the extra info.  What if we simply avoid the DO(k) macro, making DO_ROW longer.  Since it's only defined once, it's no big deal, and no change to the generated table either.  I've pushed a revert of the prior change and a fix that should work with your compiler and the generated table.  @ehsannas - please test and approve if the fix works for you.